src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -0,0 +1,329 @@
+    RAY_CHECK(false) << "Method unused";
+    Status s;
+    return s;
+  }
+  void DirectActorCallArgWaitComplete(int64_t tag) {
+    RAY_CHECK(false) << "Method unused";
+  }
+
+  // Setter, geter, and clear methods  for allocated_instances_.
+
+  void ClearAllocatedInstances() { RAY_CHECK(false) << "Method unused"; }
+
+  void ClearLifetimeAllocatedInstances() { RAY_CHECK(false) << "Method unused"; }
+
+  void SetBorrowedCPUInstances(std::vector<double> &cpu_instances) {
+    RAY_CHECK(false) << "Method unused";
+  }
+
+  std::vector<double> &GetBorrowedCPUInstances() {
+    RAY_CHECK(false) << "Method unused";
+    auto *t = new std::vector<double>();
+    return *t;
+  }
+
+  void ClearBorrowedCPUInstances() { RAY_CHECK(false) << "Method unused"; }
+
+  Task &GetAssignedTask() {
+    RAY_CHECK(false) << "Method unused";
+    auto *t = new Task();
+    return *t;
+  }
+
+  bool IsRegistered() {
+    RAY_CHECK(false) << "Method unused";
+    return false;
+  }
+
+  rpc::CoreWorkerClient *rpc_client() {
+    RAY_CHECK(false) << "Method unused";
+    return nullptr;
+  }
+
+ private:
+  WorkerID worker_id_;
+  int port_;
+  rpc::Address address_;
+  std::shared_ptr<TaskResourceInstances> allocated_instances_;
+  std::shared_ptr<TaskResourceInstances> lifetime_allocated_instances_;
+};
+
+std::shared_ptr<ClusterResourceScheduler> CreateSingleNodeScheduler(
+    const std::string &id) {
+  std::unordered_map<std::string, double> local_node_resources;
+  local_node_resources[ray::kCPU_ResourceLabel] = 8;
+  local_node_resources[ray::kGPU_ResourceLabel] = 4;
+  local_node_resources[ray::kMemory_ResourceLabel] = 128;
+
+  auto scheduler = std::make_shared<ClusterResourceScheduler>(
+      ClusterResourceScheduler(id, local_node_resources));
+
+  return scheduler;
+}
+
+Task CreateTask(const std::unordered_map<std::string, double> &required_resources) {
+  TaskSpecBuilder spec_builder;
+  TaskID id = RandomTaskId();
+  JobID job_id = RandomJobId();
+  rpc::Address address;
+  spec_builder.SetCommonTaskSpec(
+      id, Language::PYTHON, FunctionDescriptorBuilder::BuildPython("", "", "", ""),
+      job_id, TaskID::Nil(), 0, TaskID::Nil(), address, 0, required_resources, {});
+  rpc::TaskExecutionSpec execution_spec_message;
+  execution_spec_message.set_num_forwards(1);
+  return Task(spec_builder.Build(), TaskExecutionSpecification(execution_spec_message));
+}
+
+class ClusterTaskManagerTest : public ::testing::Test {
+ public:
+  void SetUp() {}
+
+  void Shutdown() {}
+};
+
+TEST_F(ClusterTaskManagerTest, SampleTest) {
+  ClientID id = ClientID::FromRandom();
+  // Task has no dependencies and shouldn't spill so these functions should never be
+  // called.
+  std::function<bool(const Task &)> fulfills_dependencies_func = nullptr;
+  std::function<boost::optional<rpc::GcsNodeInfo>(const ClientID &node_id)>
+      get_node_info_func = nullptr;
+  auto task_manager = ClusterTaskManager(id, CreateSingleNodeScheduler(id.Binary()),
```

Comment:
You might want to move these into the fixture code. Something like [this](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/test/direct_actor_transport_test.cc#L106).